src/main/java/io/kestra/plugin/aws/kinesis/PutRecords.java

```diff
@@ -0,0 +1,295 @@
+package io.kestra.plugin.aws.kinesis;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Strings;
+import io.kestra.core.exceptions.IllegalVariableEvaluationException;
+import io.kestra.core.models.annotations.Example;
+import io.kestra.core.models.annotations.Plugin;
+import io.kestra.core.models.annotations.PluginProperty;
+import io.kestra.core.models.executions.metrics.Counter;
+import io.kestra.core.models.executions.metrics.Timer;
+import io.kestra.core.models.flows.State;
+import io.kestra.core.models.tasks.RunnableTask;
+import io.kestra.core.runners.RunContext;
+import io.kestra.core.serializers.FileSerde;
+import io.kestra.core.serializers.JacksonMapper;
+import io.kestra.plugin.aws.AbstractConnection;
+import io.kestra.plugin.aws.kinesis.model.Record;
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.Flowable;
+import io.swagger.v3.oas.annotations.media.Schema;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.kinesis.KinesisClient;
+import software.amazon.awssdk.services.kinesis.KinesisClientBuilder;
+import software.amazon.awssdk.services.kinesis.model.PutRecordsRequest;
+import software.amazon.awssdk.services.kinesis.model.PutRecordsRequestEntry;
+import software.amazon.awssdk.services.kinesis.model.PutRecordsResponse;
+import software.amazon.awssdk.services.kinesis.model.PutRecordsResultEntry;
+
+import javax.validation.constraints.NotNull;
+import java.io.*;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.Duration;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import static io.kestra.core.utils.Rethrow.throwFunction;
+
+@SuperBuilder
+@ToString
+@EqualsAndHashCode
+@Getter
+@NoArgsConstructor
+@Plugin(
+    examples = {
+        @Example(
+            title = "Put multiple records as maps to Amazon Kinesis.",
+            code = {
+                "streamName: \"mystream\"",
+                "records:",
+                "  - data: \"user sign-in event\"",
+                "    explicitHashKey: \"optional hash value overriding the partition key\"",
+                "  - data: \"user sign-out event\"",
+                "    partitionKey: \"user1\""
+            }
+        ),
+        @Example(
+            title = "Put multiple records as a JSON string to Amazon Kinesis.",
+            code = {
+                "streamName: \"mystream\"",
+                "records: |",
+                "   [",
+                "     {",
+                "       \\\"Data\\\": \\\"user sign-in event\\\",",
```

Comment:
Here Data and Partition key have an uppercase, does this matter ?

Reply:
good observation, I believe yes, as this is the raw payload that will be sent to the API https://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecordsRequestEntry.html 

Let's add a link in the task documentation 👍

Reply:
Yeah it is intended as aws api use this uppdercase format and with @anna-geller we thought it would be better to support it too